line.go

```diff
@@ -0,0 +1,97 @@
+package xinvoice
+
+import (
+	"github.com/invopop/gobl/bill"
+)
+
+// Line defines the structure of the IncludedSupplyChainTradeLineItem in the CII standard
+type Line struct {
+	ID              string           `xml:"ram:AssociatedDocumentLineDocument>ram:LineID"`
+	Name            string           `xml:"ram:SpecifiedTradeProduct>ram:Name"`
+	NetPrice        string           `xml:"ram:SpecifiedLineTradeAgreement>ram:NetPriceProductTradePrice>ram:ChargeAmount"`
+	TradeDelivery   *Quantity        `xml:"ram:SpecifiedLineTradeDelivery>ram:BilledQuantity"`
+	TradeSettlement *TradeSettlement `xml:"ram:SpecifiedLineTradeSettlement"`
+}
+
+// Quantity defines the structure of the quantity with its attributes for the CII standard
+type Quantity struct {
+	Amount   string `xml:",chardata"`
+	UnitCode string `xml:"unitCode,attr"`
+}
+
+// TradeSettlement defines the structure of the SpecifiedLineTradeSettlement of the CII standard
+type TradeSettlement struct {
+	ApplicableTradeTax []*ApplicableTradeTax `xml:"ram:ApplicableTradeTax"`
+	Sum                string                `xml:"ram:SpecifiedTradeSettlementLineMonetarySummation>ram:LineTotalAmount"`
+}
+
+// ApplicableTradeTax defines the structure of ApplicableTradeTax of the CII standard
+type ApplicableTradeTax struct {
+	TaxType        string `xml:"ram:TypeCode"`
+	TaxCode        string `xml:"ram:CategoryCode"`
+	TaxRatePercent string `xml:"ram:RateApplicablePercent"`
+}
+
+func newLine(line *bill.Line) *Line {
+	if line.Item == nil {
+		return nil
+	}
+	item := line.Item
+
+	lineItem := &Line{
+		ID:       item.Name,
+		Name:     item.Name,
+		NetPrice: item.Price.String(),
+		TradeDelivery: &Quantity{
+			Amount:   line.Quantity.String(),
+			UnitCode: string(item.Unit.UNECE()),
+		},
+		TradeSettlement: newTradeSettlement(line),
+	}
+
+	return lineItem
+}
+
+func newTradeSettlement(line *bill.Line) *TradeSettlement {
+	var applicableTradeTax []*ApplicableTradeTax
+	for _, tax := range line.Taxes {
+		tradeTax := &ApplicableTradeTax{
+			TaxType: tax.Category.String(),
+			// The sales tax category codes are as follows:
+			// - S = Sales tax applies at the standard rate
+			// - Z = goods taxable at the zero rate
+			// - E = Tax exempt
+			// - AE = Reversal of tax liability
+			// - K = No sales tax is shown for intra-community deliveries
+			// - G = Tax not charged due to export outside the EU
+			// - O = Outside the tax scope
+			// - L = IGIC (Canary Islands)
+			// - M = IPSI (Ceuta/Melilla)
+			TaxCode: "S",
```

Comment:
Its great that the comment is here, but assuming `"S"` is dangerous. Is this specific for Germany? These look generic and more part of a CII standard. We can chat about this elsewhere, as its tricky.

Reply:
Which field in GOBL corresponds to sales tax category codes? :shrug:

Reply:
There is no direct mapping here, GOBL simply doesn't store data in the same way. The `L` option for example implies that the tax category is "IGIC", the `AE` implies that the invoice is tagged with `reverse-charge`. So this section will require a lot more analysis to figure out the best path.